.github/workflows/test-coding-standards.yml

```diff
@@ -34,6 +34,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Why do we need `fetch-depth: 0`?

Reply:
to fetch all branches. by default, the action will just fetch the PR branch.

Reply:
Why do we need to fetch all branches to check PR coding style?

Reply:
Unfortunately, yes. The utils composer.json is requiring `codeigniter4/codeigniter4@4.x-dev` which means it will require the `develop` branch.

If you want, I can add a workflow step to remove phpstan and phpstan-codeigniter since they won't be used in coding style.

Reply:
`fetch-depth: 0` takes too long. We should avoid it as possible.

See "Checkout" time in
https://github.com/codeigniter4/CodeIgniter4/actions/runs/10041723192/job/27750449854?pr=9047
and https://github.com/codeigniter4/CodeIgniter4/actions/runs/10054737116/job/27789901527

Reply:
Can't we remove all `fetch-depth: 0` ?

Reply:
How about moving PHPStan to the main composer.json?

Reply:
hmm, that makes sense. I remember one time that phpstan recommends to be in the main composer.json